languages/haskell/docs/implementing-a-concept-exercise.md

```diff
@@ -2,48 +2,61 @@
+- `<NAME>`: the name of the exercise in PascalCase (e.g. `MonadTransformers`).
 
-Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Concept exercises should be modelled after the CIS 194 Haskell course at Penn University, either the [Spring 2013][spring-2013] or [Fall 2016][fall-2016] version.
+Haskell covers a vast amount of concepts. Choosing a particular subset of concepts and an ideal ordering of these concepts is a contentious task, so the following approach is made: As a basis for both, the online resources for the CIS 194 course at Penn University is chosen.
 
-To implement a concept exercise, the following files need to be created:
+Contributors (you) may freely choose a concept related to Haskell and implement an exercise that covers this. Before implementing such an exercise, make sure you have a good understanding of what the exercise should be teaching (and what not). Check if this concept is already partially or fully covered by another exercise, and arrange the concept in a hierarchical fashion that corresponds to the curriculum of CIS 194.
+
+Because CIS 194 online course material is available in the [Spring 2013][spring-2013] and the [Fall 2016][fall-2016] version, and because these have different orderings, this does leave room for interpretation.
+
+Because CIS 194 may not cover all subject that contributors wish to cover in this material, extending the concepts beyond the CIS 194 course should ideally be accompanied by some level of reasoning about the curriculum.
+
+The directory structure of a single concept exercise looks like this:
 
 ```
 languages
 └── Haskell
     └── exercises
         └── concept
             └── <SLUG>
-                ├── .docs
-                |   ├── instructions.md
-                |   ├── introduction.md
-                |   ├── hints.md
-                |   └── after.md (optional)
-                ├── .meta
-                |   |── design.md
-                |   └── <SLUG>.hs (example solution)
+                ├── package.yaml
+                ├── stack.yaml
                 ├── src
-                |   └── <NAME>.hs
+                │   └── <NAME>.hs
                 ├── test
-                |   └── Tests.hs
-                ├── package.yaml
-                └── stack.yaml
+                │   └── Tests.hs
+                ├── examples
+                │   ├── success-... (default: standard)
+                │   │   ├── package.yaml
+                │   │   └── src
+                │   │       └── <NAME>.hs
+                │   └── ...
+                ├── .docs
+                │   ├── instructions.md
+                │   ├── introduction.md
+                │   ├── hints.md
+                │   └── after.md (optional)
+                └── .meta
+                    └── design.md
```

Comment:
I just noticed that I forgot to document that we re-added the `config.json` file, albeit with a different goal than before. See https://github.com/exercism/v3/pull/765. FYI: that PR also updates this document.

Reply:
Thanks, I've re-added `config.json` to the overview.